tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/SingleSaveBlobStoreDAO.java

```diff
@@ -0,0 +1,43 @@
+package com.linagora.tmail.blob.blobid.list;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStoreDAO;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.blob.api.ObjectStoreException;
+import reactor.core.publisher.Mono;
+
+import javax.inject.Inject;
+import java.util.Objects;
+
+public class SingleSaveBlobStoreDAO {
+    private final BlobStoreDAO blobStoreDAO;
+    private final BlobIdList blobIdList;
+    private final BucketName defaultBucketName;
+
+    @Inject
+    public SingleSaveBlobStoreDAO(BlobStoreDAO blobStoreDAO,
+                                  BlobIdList blobIdList,
+                                  BucketName defaultBucketName) {
+        this.blobStoreDAO = blobStoreDAO;
+        this.blobIdList = blobIdList;
+        this.defaultBucketName = defaultBucketName;
+    }
+
+    public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
+        return Mono.from(blobIdList.isStored(blobId))
+                .filter(isStored -> Objects.equals(isStored, false))
```

Comment:
```suggestion
                .filter(isStored -> !isStored)
```

Reply:
I tried it before, But I can't. `blobIdList.isStored(blobId)` return `scala.Boolean`, and `Mono.from` don't receive that
![image](https://user-images.githubusercontent.com/81145350/118214571-28f46280-b49a-11eb-86f1-d13190c06cdf.png)

Reply:
Can't we write this class in scala then?

Reply:
You might have methods for it in the Boolean object... Like `isStored.eq(false)` for example?